src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts

```diff
@@ -56,6 +65,16 @@ export abstract class DashboardPage extends AngularDisposable implements IConfig
 
 	@ViewChildren(TabChild) private _tabs: QueryList<DashboardTab>;
 	@ViewChild(PanelComponent) private _panel: PanelComponent;
+	@ViewChild('taskBar', { read: ElementRef }) private taskContainer: ElementRef;
+	protected taskbar: Taskbar;
+
+	// actions
+	protected _editAction: EditDashboardAction;
```

Comment:
this doesn't seem like a scalable way to add toolbar action.  It seems like this should be contribution driven to allow easier customization.